### ISSUE TITLE
zap2docker 2.10.0 fails with python exception "AttributeError: 'str' object has no attribute 'get' " against very specific input

### ISSUE THREAD

**Issue by ImmoStanke** [zaproxy/zaproxy#6421]
**Describe the bug**
In the latest OWASP-ZAP zap2docker-stable docker image, an unexpected failure occurs.
In very specific circumstances which I can reproduce but haven't been able to fully explain yet, `zap-baseline.py` fails with an error "AttributeError: 'str' object has no attribute 'get' " thrown in the `common.py script`. It appears to be an issue related to string escaping, as suddenly the dict is broken and is interpreted as a string by python, which causes a "get" to fail when it's trying to read the alert.
This broke our pipeline so we had to revert back to 2.9.0 version of the docker image, which works (against the same application under test).
I've narrowed the problem down to a specific vendor.js file containing dependencies, which I've uploaded to a gist so you can hopefully reproduce it. When running against the gist directly, however, the issue does not occur. See "steps to reproduce" for more details.

Although this issue is tricky to reproduce and only occurs for this specific input, I believe it's possible that it points to a bigger issue with string escaping that might need to be addressed.

```
ERROR <class 'AttributeError'>

2021-01-21 04:04:53,430 Unexpected error: <class 'AttributeError'>

Traceback (most recent call last):

  File "/zap/zap-baseline.py", line 361, in main

    alert_dict = zap_get_alerts(zap, target, ignore_scan_rules, out_of_scope_dict)

  File "/zap/zap_common.py", line 104, in _wrap

    return_data = func(*args_list, **kwargs)

  File "/zap/zap_common.py", line 490, in zap_get_alerts

    plugin_id = alert.get('pluginId')

AttributeError: 'str' object has no attribute 'get'
```

**To Reproduce**
1. Download the problematic vendors.js from here: https://gist.github.com/ImmoStanke/d227bde4c20438b4f4440c45e03166ce
2. Host it locally and make it accessible to docker (I'm on mac so I can use docker.for.mac.localhost as the url. You might need to create a separate container and put them in a network or use some other workaround to expose the file to the owasp docker container)
3. Run the zap docker image against it: `docker run -t owasp/zap2docker-stable:latest zap-baseline.py -i -d -t "http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js"` (depending on step 2 your url may vary)
4. You should be able to observe the above error near the top of the output, and the comnmand returns error code 3.

One might think that the issue would also occur when just running directly against the gist:

`docker run -t owasp/zap2docker-stable:latest zap-baseline.py -i -d -t "https://gist.githubusercontent.com/ImmoStanke/d227bde4c20438b4f4440c45e03166ce/raw/52e8fb8cb51c9d4f4f01ec837a652fa58f51eecf/vendors.ceee132f248b8cf8acfa.js"`
However, for some reason this does not cause the issue. I haven't been able to figure out why. Anyways we probably should't run ZAP against websites we don't own ;).

**Expected behavior**
The script should return exit code 0, and give a summary of non-severe issues. 
Basically, it should behave the same as
` docker run -t owasp/zap2docker-stable:2.9.0 zap-baseline.py -i -t "http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js"`

**Screenshots**

**Software versions**
 - ZAP: zap2docker:2.10.0 (latest "stable")
 - Add-on: default addons
 - OS: Docker (tested with docker-for-mac on MacOS as well as in a docker container in our pipeline which runs some form of linux as the host system)
 - Java: Whatever ships with the docker image
 - Browser: Does not apply, error occurs in command line mode only

**Errors from the zap.log file**
Full output of the script with debug flag (no other logs are available as far as I understand), this is what the command in step 3 of "how to reproduce" returns for me:
```
2021-01-27 09:42:12,066 Target: http://localhost:8080/vendors.ceee132f248b8cf8acfa.js
2021-01-27 09:42:12,066 Could not find custom hooks file at /home/zap/.zap_hooks.py 
2021-01-27 09:42:12,066 Trigger hook: cli_opts, args: 1
2021-01-27 09:42:12,066 Using port: 40753
2021-01-27 09:42:12,066 Trigger hook: start_zap, args: 2
2021-01-27 09:42:12,067 Starting ZAP
2021-01-27 09:42:12,067 Params: ['zap-x.sh', '-daemon', '-port', '40753', '-host', '0.0.0.0', '-config', 'database.recoverylog=false', '-config', 'api.disablekey=true', '-config', 'api.addrs.addr.name=.*', '-config', 'api.addrs.addr.regex=true', '-config', 'spider.maxDuration=1', '-addonupdate', '-addoninstall', 'pscanrulesBeta']
2021-01-27 09:42:12,070 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:13,073 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:14,076 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:15,079 Starting new HTTP connection (1): localhost:40753
Jan 27, 2021 9:42:16 AM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
2021-01-27 09:42:16,082 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:17,085 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:18,089 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:19,092 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:20,095 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:21,099 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:22,102 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:23,106 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:24,112 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:25,117 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:26,121 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:27,125 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:28,131 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:29,134 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:30,140 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:31,143 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:32,147 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:33,098 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:33,175 http://localhost:40753 "GET http://zap/JSON/core/view/version/ HTTP/1.1" 200 20
2021-01-27 09:42:33,176 ZAP Version 2.10.0
2021-01-27 09:42:33,176 Took 21 seconds
2021-01-27 09:42:33,176 Trigger hook: zap_started, args: 2
2021-01-27 09:42:33,176 Tune
2021-01-27 09:42:33,176 Disable all tags
2021-01-27 09:42:33,177 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:33,201 http://localhost:40753 "GET http://zap/JSON/pscan/action/disableAllTags/?apikey= HTTP/1.1" 200 15
2021-01-27 09:42:33,202 Set max pscan alerts
2021-01-27 09:42:33,203 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:33,207 http://localhost:40753 "GET http://zap/JSON/pscan/action/setMaxAlertsPerRule/?maxAlerts=10&apikey= HTTP/1.1" 200 15
2021-01-27 09:42:33,208 Trigger hook: zap_tuned, args: 1
2021-01-27 09:42:33,208 Trigger hook: zap_access_target, args: 2
2021-01-27 09:42:33,209 Starting new HTTP connection (1): localhost:40753
2021-01-27 09:42:33,239 http://localhost:40753 "GET http://localhost:8080/vendors.ceee132f248b8cf8acfa.js HTTP/1.1" 502 1966
ERROR [Errno 5] ZAP failed to access: http://localhost:8080/vendors.ceee132f248b8cf8acfa.js
2021-01-27 09:42:33,240 I/O error: [Errno 5] ZAP failed to access: http://localhost:8080/vendors.ceee132f248b8cf8acfa.js
Traceback (most recent call last):
  File "/zap/zap-baseline.py", line 331, in main
    zap_access_target(zap, target)
  File "/zap/zap_common.py", line 104, in _wrap
    return_data = func(*args_list, **kwargs)
  File "/zap/zap_common.py", line 387, in zap_access_target
    raise IOError(errno.EIO, 'ZAP failed to access: {0}'.format(target))
OSError: [Errno 5] ZAP failed to access: http://localhost:8080/vendors.ceee132f248b8cf8acfa.js
Found Java version 11.0.9.1
Available memory: 7964 MB
Using JVM args: -Xmx1991m
1002 [main] INFO  org.parosproxy.paros.Constant - Copying default configuration to /home/zap/.ZAP/config.xml
1174 [main] INFO  org.parosproxy.paros.Constant - Creating directory /home/zap/.ZAP/session
1174 [main] INFO  org.parosproxy.paros.Constant - Creating directory /home/zap/.ZAP/dirbuster
1175 [main] INFO  org.parosproxy.paros.Constant - Creating directory /home/zap/.ZAP/fuzzers
1175 [main] INFO  org.parosproxy.paros.Constant - Creating directory /home/zap/.ZAP/plugin
1301 [main] INFO  org.zaproxy.zap.DaemonBootstrap - OWASP ZAP 2.10.0 started 27/01/2021, 09:42:13 with home /home/zap/.ZAP/
1329 [main] INFO  org.parosproxy.paros.common.AbstractParam - Setting config database.recoverylog = false was null
1329 [main] INFO  org.parosproxy.paros.common.AbstractParam - Setting config api.disablekey = true was null
1330 [main] INFO  org.parosproxy.paros.common.AbstractParam - Setting config api.addrs.addr.name = .* was null
1330 [main] INFO  org.parosproxy.paros.common.AbstractParam - Setting config api.addrs.addr.regex = true was null
1330 [main] INFO  org.parosproxy.paros.common.AbstractParam - Setting config spider.maxDuration = 1 was null
1337 [main] INFO  org.parosproxy.paros.network.SSLConnector - Reading supported SSL/TLS protocols...
1337 [main] INFO  org.parosproxy.paros.network.SSLConnector - Using a SSLEngine...
1419 [main] INFO  org.parosproxy.paros.network.SSLConnector - Done reading supported SSL/TLS protocols: [SSLv2Hello, SSLv3, TLSv1, TLSv1.1, TLSv1.2, TLSv1.3]
1422 [main] INFO  org.parosproxy.paros.extension.option.OptionsParamCertificate - Unsafe SSL renegotiation disabled.
1791 [main] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache open start
1798 [main] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit start
1799 [main] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit end
1800 [main] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache open end
1845 [ZAP-daemon] INFO  org.zaproxy.zap.control.ExtensionFactory - Loading extensions
3178 [ZAP-daemon] INFO  org.zaproxy.zap.control.ExtensionFactory - Installed add-ons: [[id=alertFilters, version=10.0.0], [id=ascanrules, version=38.0.0], [id=bruteforce, version=10.0.0], [id=commonlib, version=1.2.0], [id=diff, version=10.0.0], [id=directorylistv1, version=4.0.0], [id=domxss, version=10.0.0], [id=encoder, version=0.4.0], [id=formhandler, version=3.0.0], [id=fuzz, version=13.1.0], [id=gettingStarted, version=12.0.0], [id=graaljs, version=0.1.0], [id=graphql, version=0.2.0], [id=help, version=11.0.0], [id=hud, version=0.12.0], [id=importurls, version=7.0.0], [id=invoke, version=10.0.0], [id=onlineMenu, version=8.0.0], [id=openapi, version=17.0.0], [id=pscanrules, version=31.0.0], [id=quickstart, version=29.0.0], [id=replacer, version=8.0.0], [id=retire, version=0.6.0], [id=reveal, version=3.0.0], [id=saverawmessage, version=5.0.0], [id=savexmlmessage, version=0.1.0], [id=scripts, version=27.0.0], [id=selenium, version=15.3.0], [id=soap, version=4.0.0], [id=spiderAjax, version=23.2.0], [id=tips, version=7.0.0], [id=webdriverlinux, version=23.0.0], [id=websocket, version=22.0.0], [id=zest, version=33.0.0]]
3689 [ZAP-daemon] INFO  org.zaproxy.zap.control.ExtensionFactory - Extensions loaded
3900 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows ZAP to check for updates
3903 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Options Extension
3903 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Edit Menu Extension
3903 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Provides a rest based API for controlling and accessing ZAP
3910 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Session State Extension
3911 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Report Extension
3911 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing History Extension
3912 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Show hidden fields and enable disabled fields
3913 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Search messages for strings and regular expressions
3914 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to intercept and modify requests and responses
3916 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Passive scanner
3952 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Script Passive Scan Rules
3952 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Stats Passive Scan Rule
3953 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Vulnerable JS Library
3953 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: WSDL File Detection
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Application Error Disclosure
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Incomplete or No Cache-control and Pragma HTTP Header Set
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Charset Mismatch
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: CSP
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Content-Type Header Missing
3954 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie No HttpOnly Flag
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Loosely Scoped Cookie
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie Without SameSite Attribute
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie Without Secure Flag
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cross-Domain Misconfiguration
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cross-Domain JavaScript Source File Inclusion
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Absence of Anti-CSRF Tokens
3955 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Private IP Disclosure
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Session ID in URL Rewrite
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Debug Error Messages
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Sensitive Information in URL
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Sensitive Information in HTTP Referrer Header
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Suspicious Comments
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Weak Authentication Method
3956 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Insecure JSF ViewState
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Secure Pages Include Mixed Content
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Timestamp Disclosure
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Username Hash Found
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Viewstate
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-AspNet-Version Response Header
3957 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Content-Type-Options Header Missing
3958 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Debug-Token Information Leak
3958 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Frame-Options Header
3958 [ZAP-daemon] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Server Leaks Information via "X-Powered-By" HTTP Response Header Field(s)
3981 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to view and manage alerts
3983 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Active scanner, heavily based on the original Paros active scanner, but with additional tests added
3989 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Spider used for automatically finding URIs on a site
3993 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing A set of common popup menus for miscellaneous tasks
3994 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Forced browsing of files and directories using code from the OWASP DirBuster tool
3995 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Manual Request Editor Extension
3995 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Compares 2 sessions and generates an HTML file showing the differences
3996 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Invoke external applications passing context related information such as URLs and parameters
3996 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Handles anti cross site request forgery (CSRF) tokens
4000 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Authentication Extension
4017 [ZAP-daemon] INFO  org.zaproxy.zap.extension.authentication.ExtensionAuthentication - Loaded authentication method types: [Form-based Authentication, HTTP/NTLM Authentication, Manual Authentication, Script-based Authentication, JSON-based Authentication]
4035 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Creates a dynamic SSL certificate to allow SSL communications to be intercepted without warnings being generated by the browser
4035 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Logs errors to the Output tab in development mode only
4035 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Users Extension
4038 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Summarise and analyse FORM and URL parameters as well as cookies
4039 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Script integration
4043 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Scripting console, supports all JSR 223 scripting languages
4161 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Forced User Extension
4162 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Extension handling HTTP sessions
4164 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Zest is a specialized scripting language, originally, from Mozilla specifically designed to be used in security tools
4408 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing ExtensionDiff
4408 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Post Table View Extension
4409 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Adds support for scriptable encoders to ZAP.
4410 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Session Management Extension
4419 [ZAP-daemon] INFO  org.zaproxy.zap.extension.sessions.ExtensionSessionManagement - Loaded session management method types: [Cookie-based Session Management, HTTP Authentication Session Management, Script-based Session Management]
4421 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Form Table View Extension
4421 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Capture messages from WebSockets with the ability to set breakpoints.
4450 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to import a file containing URLs which ZAP will access, adding them to the Sites tree
4453 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to import a WSDL file containing operations which ZAP will access, adding them to the Sites tree.
4455 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Core UI related functionality.
4455 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Authorization Extension
4456 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing AJAX Spider, uses Crawljax
4458 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Provides WebDrivers to control several browsers using Selenium and includes HtmlUnit browser.
4467 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Manages the local proxy configurations
4467 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Handles adding Global Excluded URLs
4468 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Adds menu item to refresh the Sites tree
4468 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing OWASP ZAP User Guide
4468 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Provides a URL suitable for calling from target sites
4470 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to configure which extensions are loaded when ZAP starts
4470 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Combined HTTP Panels Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Hex View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Image View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Large Request View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Large Response View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Query Table View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing HTTP Panel Syntax Highlighter View Extension
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Adds support for configurable keyboard shortcuts for all of the ZAP menus.
4471 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Active and passive rule configuration
4475 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Statistics
4476 [ZAP-daemon] INFO  org.zaproxy.zap.extension.stats.ExtensionStats - Start recording in memory stats
4477 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Custom Pages Definition
4478 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to inspect and attack GraphQL endpoints.
4484 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing This extension allows a user to change the default values used by ZAP Spiders.
4487 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Active Scan Rules
4488 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing The ZAP Getting Started Guide
4489 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Easy way to replace strings in requests and responses
4494 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing ExtensionSaveXMLHttpMessage
4495 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows to fuzz WebSocket messages.
4495 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Tips and Tricks
4496 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Provides the foundation for concrete message types (for example, HTTP, WebSockets) expose fuzzer implementations.
4497 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows to fuzz HTTP messages.
4498 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Provides the GraalVM JavaScript engine for ZAP scripting.
4809 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Context alert rules filter
4814 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Heads Up Display
4882 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing ExtensionHUDlaunch
4884 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing ExtensionSaveRawHttpMessage
4885 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Passive Scan Rules
4885 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing DOM XSS Active Scan Rule
4928 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Adds the Quick Start panel for scanning and exploring applications
4931 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Add the option to use the Ajax Spider in the Quick Start scan
4932 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Launch browsers proxying through ZAP
4933 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Launch browsers proxying through ZAP
4943 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing Allows you to spider and import OpenAPI (Swagger) definitions 
4975 [ZAP-daemon] INFO  org.parosproxy.paros.extension.ExtensionLoader - Initializing The Online menu links
4981 [ZAP-daemon] INFO  org.zaproxy.zap.extension.callback.ExtensionCallback - Started callback server on 0.0.0.0:46335
4985 [ZAP-daemon] INFO  org.zaproxy.zap.extension.dynssl.ExtensionDynSSL - Creating new root CA certificate
5656 [ZAP-daemon] INFO  org.zaproxy.zap.extension.dynssl.ExtensionDynSSL - New root CA certificate created
6668 [ZAP-daemon] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - There is/are 5 newer addons
18632 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon pscanrules v32.0.0
18682 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Application Error Disclosure
18682 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Incomplete or No Cache-control and Pragma HTTP Header Set
18682 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Charset Mismatch
18682 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: CSP
18682 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Content-Type Header Missing
18683 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie No HttpOnly Flag
18683 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Loosely Scoped Cookie
18683 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie Without SameSite Attribute
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie Without Secure Flag
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cross-Domain Misconfiguration
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cross-Domain JavaScript Source File Inclusion
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Absence of Anti-CSRF Tokens
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Private IP Disclosure
18684 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Session ID in URL Rewrite
18685 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Debug Error Messages
18685 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Sensitive Information in URL
18685 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Sensitive Information in HTTP Referrer Header
18685 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Information Disclosure - Suspicious Comments
18685 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Weak Authentication Method
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Insecure JSF ViewState
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Secure Pages Include Mixed Content
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Timestamp Disclosure
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Username Hash Found
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Viewstate
18686 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-AspNet-Version Response Header
18687 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Content-Type-Options Header Missing
18687 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Debug-Token Information Leak
18687 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Frame-Options Header
18687 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Server Leaks Information via "X-Powered-By" HTTP Response Header Field(s)
18695 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon pscanrules v32.0.0
18723 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon scripts v28.0.0
19117 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon scripts v28.0.0
19130 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon soap v5.0.0
19184 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: WSDL File Detection
19188 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon soap v5.0.0
19315 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon websocket v23.0.0
19482 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon websocket v23.0.0
19492 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon webdriverlinux v24.0.0
19727 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon webdriverlinux v24.0.0
19734 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/pscanrules-release-32.zap
19735 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/scripts-beta-28.zap
19735 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/webdriverlinux-release-24.zap
19735 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/websocket-release-23.zap
19735 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/soap-alpha-5.zap
19735 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on update check complete
19737 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Downloading add-on from: https://github.com/zaproxy/zap-extensions/releases/download/pscanrulesBeta-v24/pscanrulesBeta-beta-24.zap
20594 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon pscanrulesBeta v24.0.0
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Big Redirect Detected (Potential Sensitive Information Leak)
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Content Security Policy (CSP) Header Not Set
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Directory Browsing
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Hash Disclosure
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Heartbleed OpenSSL Vulnerability (Indicative)
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: HTTP to HTTPS Insecure Transition in Form Post
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: HTTPS to HTTP Insecure Transition in Form Post
20620 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Reverse Tabnabbing
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Modern Web Application
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: PII Disclosure
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Retrieved from Cache
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: HTTP Server Response Header
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: HTTP Parameter Override
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Strict-Transport-Security Header
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: User Controllable Charset
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Cookie Poisoning
20621 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: User Controllable HTML Element Attribute (Potential XSS)
20622 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: User Controllable JavaScript Event (XSS)
20622 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Open Redirect
20622 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-Backend-Server Header Information Leak
20622 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: X-ChromeLogger-Data (XCOLD) Header Information Leak
20624 [ZAP-DownloadInstaller] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon pscanrulesBeta v24.0.0
20686 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/pscanrules-release-32.zap
20686 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/scripts-beta-28.zap
20687 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/webdriverlinux-release-24.zap
20687 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/websocket-release-23.zap
20687 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/soap-alpha-5.zap
20687 [ZAP-daemon] INFO  org.parosproxy.paros.CommandLine - Add-on downloaded to: /home/zap/.ZAP/plugin/pscanrulesBeta-beta-24.zap
20688 [ZAP-daemon] INFO  org.zaproxy.zap.DaemonBootstrap - ZAP is now listening on 0.0.0.0:40753
2021-01-27 09:42:33,245 Trigger hook: pre_exit, args: 3
```

**Additional context**
The following log lines inserted into common.py might be helpful:
```
@hook(wrap=True)
def zap_get_alerts(zap, baseurl, ignore_scan_rules, out_of_scope_dict):
    # Retrieve the alerts using paging in case there are lots of them
    st = 0
    pg = 5000
    alert_dict = {}
    alert_count = 0
    alerts = zap.core.alerts(baseurl=baseurl, start=st, count=pg)
    while len(alerts) > 0:
        logging.debug('Reading ' + str(pg) + ' alerts from ' + str(st))
        alert_count += len(alerts)
        for alert in alerts:
            logging.debug('Alert type is: ' + str(type(alert)))
            if (type(alert) == type("string")):
                logging.debug("String typed alert is: " + str(alert))
            plugin_id = alert.get('pluginId')
            if plugin_id in ignore_scan_rules:
                continue
            if not is_in_scope(plugin_id, alert.get('url'), out_of_scope_dict):
                continue
            if alert.get('risk') == 'Informational':
                # Ignore all info alerts - some of them may have been downgraded by security annotations
                continue
            if (plugin_id not in alert_dict):
                alert_dict[plugin_id] = []
            alert_dict[plugin_id].append(alert)
        st += pg
        alerts = zap.core.alerts(start=st, count=pg)
    logging.debug('Total number of alerts: ' + str(alert_count))
    return alert_dict
```

When ran a modified dockerfile with those debug lines inserted, the output was:
```
zap@a69847f98e79:/zap$ zap-baseline.py -d -t "http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js"
2021-01-27 08:46:24,909 Target: http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js
2021-01-27 08:46:24,909 Could not find custom hooks file at /home/zap/.zap_hooks.py 
2021-01-27 08:46:24,909 Trigger hook: cli_opts, args: 1
2021-01-27 08:46:24,909 Using port: 60718
2021-01-27 08:46:24,909 Trigger hook: start_zap, args: 2
2021-01-27 08:46:24,909 Starting ZAP
2021-01-27 08:46:24,909 Params: ['zap-x.sh', '-daemon', '-port', '60718', '-host', '0.0.0.0', '-config', 'database.recoverylog=false', '-config', 'api.disablekey=true', '-config', 'api.addrs.addr.name=.*', '-config', 'api.addrs.addr.regex=true', '-config', 'spider.maxDuration=1', '-addonupdate', '-addoninstall', 'pscanrulesBeta']
2021-01-27 08:46:24,913 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:25,915 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:26,917 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:27,920 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:28,923 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:29,926 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:30,931 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:31,936 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:32,003 http://localhost:60718 "GET http://zap/JSON/core/view/version/ HTTP/1.1" 200 20
2021-01-27 08:46:32,004 ZAP Version 2.10.0
2021-01-27 08:46:32,004 Took 7 seconds
2021-01-27 08:46:32,004 Trigger hook: zap_started, args: 2
2021-01-27 08:46:32,004 Tune
2021-01-27 08:46:32,004 Disable all tags
2021-01-27 08:46:32,006 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:32,010 http://localhost:60718 "GET http://zap/JSON/pscan/action/disableAllTags/?apikey= HTTP/1.1" 200 15
2021-01-27 08:46:32,011 Set max pscan alerts
2021-01-27 08:46:32,012 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:32,016 http://localhost:60718 "GET http://zap/JSON/pscan/action/setMaxAlertsPerRule/?maxAlerts=10&apikey= HTTP/1.1" 200 15
2021-01-27 08:46:32,017 Trigger hook: zap_tuned, args: 1
2021-01-27 08:46:32,017 Trigger hook: zap_access_target, args: 2
2021-01-27 08:46:32,018 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:32,095 http://localhost:60718 "GET http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js HTTP/1.1" 200 713479
2021-01-27 08:46:34,104 Trigger hook: zap_spider, args: 2
2021-01-27 08:46:34,105 Spider http://docker.for.mac.localhost:8080/
2021-01-27 08:46:34,107 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:34,133 http://localhost:60718 "GET http://zap/JSON/spider/action/scan/?apikey=&url=http%3A%2F%2Fdocker.for.mac.localhost%3A8080%2F HTTP/1.1" 200 12
2021-01-27 08:46:39,106 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:39,108 http://localhost:60718 "GET http://zap/JSON/spider/view/status/?scanId=0 HTTP/1.1" 200 16
2021-01-27 08:46:39,109 Spider complete
2021-01-27 08:46:39,109 Trigger hook: zap_spider_wrap, args: 1
2021-01-27 08:46:39,110 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:39,112 http://localhost:60718 "GET http://zap/JSON/pscan/view/recordsToScan/ HTTP/1.1" 200 21
2021-01-27 08:46:39,113 Records to scan...
2021-01-27 08:46:39,114 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:39,118 http://localhost:60718 "GET http://zap/JSON/pscan/view/recordsToScan/ HTTP/1.1" 200 21
2021-01-27 08:46:39,119 Passive scanning complete
2021-01-27 08:46:39,120 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:39,124 http://localhost:60718 "GET http://zap/JSON/core/view/urls/ HTTP/1.1" 200 261
Total of 5 URLs
2021-01-27 08:46:39,125 Trigger hook: zap_get_alerts, args: 4
2021-01-27 08:46:39,126 Starting new HTTP connection (1): localhost:60718
2021-01-27 08:46:39,432 http://localhost:60718 "GET http://zap/JSON/core/view/alerts/?baseurl=http%3A%2F%2Fdocker.for.mac.localhost%3A8080%2F&start=0&count=5000 HTTP/1.1" 200 769736
2021-01-27 08:46:39,440 Reading 5000 alerts from 0
2021-01-27 08:46:39,441 Alert type is: <class 'dict'>
2021-01-27 08:46:39,441 Alert type is: <class 'dict'>
2021-01-27 08:46:39,441 Alert type is: <class 'dict'>
2021-01-27 08:46:39,441 Alert type is: <class 'dict'>
2021-01-27 08:46:39,441 Alert type is: <class 'str'>
2021-01-27 08:46:39,441 String typed alert is: {"sourceid":"3","other":"The following pattern was used: \\bDB\\b, see evidence field for the suspicious comment/snippet.","method":"GET","evidence":function(){ var e=E();return this.copyTo(e),e},x.prototype.intValue=function(){if(this.s<0){if(1==this.t)return this[0]-this.DV;if(0==this.t)return-1}else{if(1==this.t)return this[0];if(0==this.t)return 0}return(this[1]&(1<<32-this.DB)-1)<<this.DB|this[0]},x.prototype.byteValue=function(){return 0==this.t?this.s:this[0]<<24>>24},x.prototype.shortValue=function(){return 0==this.t?this.s:this[0]<<16>>16},x.prototype.signum=function(){return this.s<0?-1:this.t<=0||1==this.t&&this[0]<=0?0:1},x.prototype.toByteArray=function(){var e=this.t,t=new Array;t[0]=this.s;var n,r=this.DB-e*this.DB%8,o=0;if(e-- >0)for(r<this.DB&&(n=this[e]>>r)!=(this.s&this.DM)>>r&&(t[o++]=n|this.s<<this.DB-r);e>=0;)r<8?(n=(this[e]&(1<<r)-1)<<8-r,n|=this[--e]>>(r+=this.DB-8)):(n=this[e]>>(r-=8)&255,r<=0&&(r+=this.DB,--e)),0!=(128&n)&&(n|=-256),0==o&&(128&this.s)!=(128&n)&&++o,(o>0||n!=this.s)&&(t[o++]=n);return t},x.prototype.equals=function(e){return 0==this.compareTo(e)},x.prototype.min=function(e){return this.compareTo(e)<0?this:e},x.prototype.max=function(e){return this.compareTo(e)>0?this:e},x.prototype.and=function(e){var t=E();return this.bitwiseTo(e,j,t),t},x.prototype.or=function(e){var t=E();return this.bitwiseTo(e,L,t),t},x.prototype.xor=function(e){var t=E();return this.bitwiseTo(e,N,t),t},x.prototype.andNot=function(e){var t=E();return this.bitwiseTo(e,M,t),t},x.prototype.not=function(){for(var e=E(),t=0;t<this.t;++t)e[t]=this.DM&~this[t];return e.t=this.t,e.s=~this.s,e},x.prototype.shiftLeft=function(e){var t=E();return e<0?this.rShiftTo(-e,t):this.lShiftTo(e,t),t},x.prototype.shiftRight=function(e){var t=E();return e<0?this.lShiftTo(-e,t):this.rShiftTo(e,t),t},x.prototype.getLowestSetBit=function(){for(var e=0;e<this.t;++e)if(0!=this[e])return e*this.DB+D(this[e]);return this.s<0?this.t*this.DB:-1},x.prototype.bitCount=function(){for(var e=0,t=this.s&this.DM,n=0;n<this.t;++n)e+=B(this[n]^t);return e},x.prototype.testBit=function(e){var t=Math.floor(e/this.DB);return t>=this.t?0!=this.s:0!=(this[t]&1<<e%this.DB)},x.prototype.setBit=function(e){return this.changeBit(e,L)},x.prototype.clearBit=function(e){return this.changeBit(e,M)},x.prototype.flipBit=function(e){return this.changeBit(e,N)},x.prototype.add=function(e){var t=E();return this.addTo(e,t),t},x.prototype.subtract=function(e){var t=E();return this.subTo(e,t),t},x.prototype.multiply=function(e){var t=E();return this.multiplyTo(e,t),t},x.prototype.divide=function(e){var t=E();return this.divRemTo(e,t,null),t},x.prototype.remainder=function(e){var t=E();return this.divRemTo(e,null,t),t},x.prototype.divideAndRemainder=function(e){var t=E(),n=E();return this.divRemTo(e,t,n),new Array(t,n)},x.prototype.modPow=function(e,t){var n,r,o=e.bitLength(),i=T(1);if(o<=0)return i;n=o<18?1:o<48?3:o<144?4:o<768?5:6,r=o<8?new I(t):t.isEven()?new V(t):new R(t);var a=new Array,s=3,u=n-1,c=(1<<n)-1;if(a[1]=r.convert(this),n>1){var l=E();for(r.sqrTo(a[1],l);s<=c;)a[s]=E(),r.mulTo(l,a[s-2],a[s]),s+=2}var f,d,h=e.t-1,p=!0,g=E();for(o=A(e[h])-1;h>=0;){for(o>=u?f=e[h]>>o-u&c:(f=(e[h]&(1<<o+1)-1)<<u-o,h>0&&(f|=e[h-1]>>this.DB+o-u)),s=n;0==(1&f);)f>>=1,--s;if((o-=s)<0&&(o+=this.DB,--h),p)a[f].copyTo(i),p=!1;else{for(;s>1;)r.sqrTo(i,g),r.sqrTo(g,i),s-=2;s>0?r.sqrTo(i,g):(d=i,i=g,g=d),r.mulTo(g,a[f],i)}for(;h>=0&&0==(e[h]&1<<o);)r.sqrTo(i,g),d=i,i=g,g=d,--o<0&&(o=this.DB-1,--h)}return r.revert(i)},x.prototype.modInverse=function(e){var t=e.isEven();if(this.isEven()&&t||0==e.signum())return x.ZERO;for(var n=e.clone(),r=this.clone(),o=T(1),i=T(0),a=T(0),s=T(1);0!=n.signum();){for(;n.isEven();)n.rShiftTo(1,n),t?(o.isEven()&&i.isEven()||(o.addTo(this,o),i.subTo(e,i)),o.rShiftTo(1,o)):i.isEven()||i.subTo(e,i),i.rShiftTo(1,i);for(;r.isEven();)r.rShiftTo(1,r),t?(a.isEven()&&s.isEven()||(a.addTo(this,a),s.subTo(e,s)),a.rShiftTo(1,a)):s.isEven()||s.subTo(e,s),s.rShiftTo(1,s);n.compareTo(r)>=0?(n.subTo(r,n),t&&o.subTo(a,o),i.subTo(s,i)):(r.subTo(n,r),t&&a.subTo(o,a),s.subTo(i,s))}return 0!=r.compareTo(x.ONE)?x.ZERO:s.compareTo(e)>=0?s.subtract(e):s.signum()<0?(s.addTo(e,s),s.signum()<0?s.add(e):s):s},x.prototype.pow=function(e){return this.exp(e,new U)},x.prototype.gcd=function(e){var t=this.s<0?this.negate():this.clone(),n=e.s<0?e.negate():e.clone();if(t.compareTo(n)<0){var r=t;t=n,n=r}var o=t.getLowestSetBit(),i=n.getLowestSetBit();if(i<0)return t;for(o<i&&(i=o),i>0&&(t.rShiftTo(i,t),n.rShiftTo(i,n));t.signum()>0;)(o=t.getLowestSetBit())>0&&t.rShiftTo(o,t),(o=n.getLowestSetBit())>0&&n.rShiftTo(o,n),t.compareTo(n)>=0?(t.subTo(n,t),t.rShiftTo(1,t)):(n.subTo(t,n),n.rShiftTo(1,n));return i>0&&n.lShiftTo(i,n),n},x.prototype.isProbablePrime=function(e){var t,n=this.abs();if(1==n.t&&n[0]<=W[W.length-1]){for(t=0;t<W.length;++t)if(n[0]==W[t])return!0;return!1}if(n.isEven())return!1;for(t=1;t<W.length;){for(var r=W[t],o=t+1;o<W.length&&r<G;)r*=W[o++];for(r=n.modInt(r);t<o;)if(r%W[t++]==0)return!1}return n.millerRabin(e)},x.prototype.square=function(){var e=E();return this.squareTo(e),e},J.prototype.init=function(e){var t,n,r;for(t=0;t<256;++t)this.S[t]=t;for(n=0,t=0;t<256;++t)n=n+this.S[t]+e[t%e.length]&255,r=this.S[t],this.S[t]=this.S[n],this.S[n]=r;this.i=0,this.j=0},J.prototype.next=function(){var e;return this.i=this.i+1&255,this.j=this.j+this.S[this.i]&255,e=this.S[this.i],this.S[this.i]=this.S[this.j],this.S[this.j]=e,this.S[e+this.S[this.i]&255]},null==q){var Y;if(q=new Array,z=0,void 0!==o&&(void 0!==o.crypto||void 0!==o.msCrypto)){var $=o.crypto||o.msCrypto;if($.getRandomValues){var Z=new Uint8Array(32);for($.getRandomValues(Z),Y=0;Y<32;++Y)q[z++]=Z[Y]}else if("Netscape"==r.appName&&r.appVersion<"5"){var Q=o.crypto.random(32);for(Y=0;Y<Q.length;++Y)q[z++]=255&Q.charCodeAt(Y)}}for(;z<256;)Y=Math.floor(65536*Math.random()),q[z++]=Y>>>8,q[z++]=255&Y;z=0,X()}function ee(){if(null==K){for(X(),(K=new J).init(q),z=0;z<q.length;++z)q[z]=0;z=0}return K.next()}function te(){ },"pluginId":"10027","cweid":"200","confidence":"Low","wascid":"13","description":"The response appears to contain suspicious comments which may help an attacker. Note: Matches made within script blocks or files are against the entire content not only comments.","messageId":"1","url":"http://docker.for.mac.localhost:8080/vendors.ceee132f248b8cf8acfa.js","reference":"","solution":"Remove all comments that return information that may help an attacker and fix any underlying problems they refer to.","alert":"Information Disclosure - Suspicious Comments","param":"","attack":"","name":"Information Disclosure - Suspicious Comments","risk":"Informational","id":"4","alertRef":"10027"}
ERROR <class 'AttributeError'>
2021-01-27 08:46:39,442 Unexpected error: <class 'AttributeError'>
Traceback (most recent call last):
  File "/zap/zap-baseline.py", line 361, in main
    alert_dict = zap_get_alerts(zap, target, ignore_scan_rules, out_of_scope_dict)
  File "/zap/zap_common.py", line 104, in _wrap
    return_data = func(*args_list, **kwargs)
  File "/zap/zap_common.py", line 493, in zap_get_alerts
    plugin_id = alert.get('pluginId')
AttributeError: 'str' object has no attribute 'get'
Found Java version 11.0.9.1
Available memory: 7964 MB
Using JVM args: -Xmx1991m
1140 [main] INFO  org.zaproxy.zap.DaemonBootstrap - OWASP ZAP 2.10.0 started 27/01/2021, 08:46:26 with home /home/zap/.ZAP/
[... more uninteresting logs...]

```

**Would you like to help fix this issue?**
Since this is a bit tricky to reproduce I'm happy to advise there. Can also provide more context if you have specific questions. Beyond that, I'm a bit lost and wouldn't know where to start/continue looking into this.

**Comment by psiinon** [comment]
Thank you for the very detailed bug report and steps to reproduce.
I agree that it probably an underlying escaping issue that we will definitely want to fix.

**Comment by psiinon** [comment]
I can confirm that thanks to your excellent reporting I can reproduce this error :D

**Comment by psiinon** [comment]
Initial findings:
The problem alert is: 10027 Information Disclosure - Suspicious Comments
The regex pattern its using is Regex pattern \\bDB\\b - 'DB' does appear as a word but it looks safe so this looks like a false positive
The evidence includes lots more than just 'DB' so there may be a problem with the evidence gathered by the rule, but thats a separate issue.

The raw alert starts with:
`alert {"sourceid":"3","other":"The following pattern was used: \\bDB\\b, see evidence field for the suspicious comment/snippet.","method":"GET","evidence":function(){ var e=E();`

Note that the evidence value does not start with a double quote.
When read directly from the API the alerts all look ok - there are actually over a 100 for just this one file (!) so I havnt checked them all in detail yet, but the JSON looked sane when viewed in Firefox.

My working assumption is that this could be a problem with our python library treating the evidence value starting with 'function(){...' as a function rather than a string. But I'll need to look into this further.
There will be a variety of ways in which this problem could be avoided, from disabling that scan rule to excluding that specific file, but we do need to fix this asap...

**Comment by psiinon** [comment]
OK, so this _is_ a core issue, but it occurs on 2.10 rather than 2.9 as a side effect of the recent changes to the 10027 Information Disclosure - Suspicious Comments rule as per https://github.com/zaproxy/zap-extensions/commit/161ba43886b263a7e875cd2929411050fd01a19a
In specific cases the evidence is really not helpful so we should fix that which should avoid the problem in this case, but obviously we will want to fix the underlying JSON encoding problem as well.

**Comment by psiinon** [comment]
I now have a much easier was to reproduce this issue :)
Just manually create an alert with evidence: `function(){}`
The problem code is likely to be here: https://github.com/zaproxy/zaproxy/blob/main/zap/src/main/java/org/zaproxy/zap/utils/JsonUtil.java#L47-L52

I have a fix for this specific case, but no a huge amount of confidence that it will solve all of the possible problems :/

**Comment by psiinon** [comment]
Its worth noting that some whitespace seems to be ignored, so we'll need to check using a regex. And check the underlying library to see what else it auto converts :/

**Comment by psiinon** [comment]
@ImmoStanke FYI the bug is actually present in ZAP 2.9.0 as well. The reason you only experience it on 2.10.0 is that it was exposed by a change to one of the passive scan rules, and that change was only made available on 2.10.0.
I've raised a PR to change the info included for these alerts which will have the side effect of avpoiding this core issue in this specific case: https://github.com/zaproxy/zap-extensions/pull/2703 
Once this PR is merged and the passive scan rules released then you will be able to go back to using 2.10.0.
There is still the possibility that this bug will be triggered on 2.9.0 and 2.10.0 but hopefully it will be much less likely to occur.
If people keep on hitting it then we may have to generate as bug fix release, but I'm hoping that wont be necessary.
Note that the next weekly will have a fix in it for this bug.

**Comment by psiinon** [comment]
@ImmoStanke the pscanrules add-on has been released so you can go back to using ZAP 2.10.0, just make sure you update all of the add-ons :)

**Comment by ImmoStanke** [comment]
Can confirm that after pulling a fresh build of the latest image, the issue is resolved for me. Thanks a lot for the quick fix!
